versioned_docs/version-1.0/cadence-migration-guide/usdc-migration.mdx

```diff
@@ -0,0 +1,330 @@
+            signer.save(
+                <-wrappedTokens,
+                to: USDCFlow.VaultStoragePath
+            )
+
+            // Set up the correct capabilities
+            signer.link<&USDCFlow.Vault{FungibleToken.Receiver}>(
+                USDCFlow.ReceiverPublicPath,
+                target: USDCFlow.VaultStoragePath
+            )
+            signer.link<&USDCFlow.Vault{FungibleToken.Balance, MetadataViews.Resolver}>(
+                USDCFlow.VaultPublicPath,
+                target: USDCFlow.VaultStoragePath
+            )
+        }
+    }
+}
+```
+
+You could have some similar but slightly more complex code to move
+the code from Vaults stored in a dictionary or an array.
+
+### Vault Field Stored in a Composite Type
+
+A `FiatToken.Vault` field stored in a composite type like a resource is a bit more complex.
+Example:
+```cadence
+pub contract VaultInResource {
+
+    pub resource ResourceWithVault {
+        pub let vault: @FiatToken.Vault
+    }
+}
+```
+
+It is more challenging to deal with vaults that are stored in a composite type
+becuase those can live in any account. Fixing them will be more difficult because
+you have to work with users to change it. 
+
+Luckily, this is a very rare pattern and might not exist anywhere,
+so hopefully it won't be an issue.
+
+In case it is an issue for someone, please reach out to the Flow team for assistance
+and the team can likely find a way to help you resolve this.
+
+### Capability to a `FiatToken.Vault`
+
+Any contracts with capability fields that point to `FiatToken` vaults
+will need to add a way to redirect that capability to the `USDCFlow` vault in
+the same account as the `FiatToken` capability.
+
+This could be done by checking the owner address of the vault that capability points to
+and getting the `USDCFlow` vault capability from that address
+instead of using the existing `FiatToken` capability.
+
+If the capability is a `FungibleToken.Provider` capability, then you may have to
+create a new type that stores the `USDCFlow` capability since you can't
+recreate the provider capability without the owner creating a new one themselves.
+
+Either way, again you'll need to communicate with your users as much as possible
+to convince them to swap their `FiatToken` to `USDCFlow`
+to be compatible with the new tokens and capabilities.
+
+## NFTStorefront with `FiatToken` Listings
+
+As described above in the user section, all `NFTStorefront` listings that
+expect `FiatToken` to be used as the form of payment will break after the Crescendo upgrade.
+
+Any app or project that is using `NFTStorefront` or a marketplace that is similar with USDC
+should just expect all `FiatToken` listings to break and should expect users to just
+create new listings that use `USDCFlow` instead. These listings that break
+will not cause any assets to be lost or compromised. It will just require a bit of work
+to re-list all the assets that were listed before.
```

Comment:
```suggestion
Any app or project using `NFTStorefront` or a similar `USDC`-based marketplace should anticipate that all `FiatToken` listings will cease to function. Users will need to create new listings using `USDCFlow` instead. It's important to note that while these broken listings won't cause any assets to be lost or compromised, it will require some effort to re-list all previously listed assets.
```